### PR TITLE
feat(Masthead): Updates for new masthead structure in Penta

### LINKED ISSUE #10788: Masthead - updates to support better logo layout
Placeholder issue for https://github.com/patternfly/patternfly/pull/6870

So far this update will just be to add a new container around the masthead toggle and logo, and rename a couple of containers within it.

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #10788 

- Renamed `MastheadBrand` to `MastheadLogo`
- Renamed `MastheadMain` to `MastheadBrand`
- Wrap `MastheadToggle` and `MastheadBrand` in `MastheadMain`

Codemod issue: https://github.com/patternfly/pf-codemods/issues/718

### CODE REVIEW

**Review by wise-king-sullyman — APPROVED** [review]

**Review by srambach — COMMENTED** [review]
Surge is down so I haven't run it yet but had a couple of questions. Otherwise looks good on inspection!

File: packages/react-core/src/components/Masthead/examples/Masthead.md [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
 
 import BarsIcon from '@patternfly/react-icons/dist/js/icons/bars-icon';
 import { Link } from '@reach/router';
 import pfIcon from '../../assets/PF-HorizontalLogo-Color.svg';
 
-`Masthead` should contain the following components to maintain proper layout and formatting: `MastheadToggle`, `MastheadMain`, and `MastheadContent`.
+`Masthead` should contain the following components to maintain proper layout and formatting: `MastheadMain`, and `MastheadContent`.
 
-`MastheadMain` represents the smaller area taken up by a logo, and will typically contain a `MastheadBrand`. `MastheadContent` represents the main portion of the masthead area and will typically contain a `Toolbar` or other menu-like content such as `Dropdown`.
+`MastheadMain` represents the smaller area taken up by a logo, and will typically contain a `MastheadLogo`. `MastheadContent` represents the main portion of the masthead area and will typically contain a `Toolbar` or other menu-like content such as `Dropdown`.
```
[the remark this answers is not in the document]
Comment (tlabaj) [reply packages/react-core/src/components/Masthead/examples/Masthead.md:14]: Yes!  I meant to go back an look at the wording.  I will see if Erin can review it too.

File: packages/react-core/src/components/Masthead/examples/Masthead.md [hunk 2]
```diff
@@ -2,16 +2,16 @@
...
-propComponents: ['Masthead', 'MastheadToggle', 'MastheadMain', 'MastheadBrand', 'MastheadContent']
+propComponents: ['Masthead', 'MastheadToggle', 'MastheadMain', 'MastheadBrand', MastheadLogo, 'MastheadContent']
 ---
 
 import BarsIcon from '@patternfly/react-icons/dist/js/icons/bars-icon';
 import { Link } from '@reach/router';
 import pfIcon from '../../assets/PF-HorizontalLogo-Color.svg';
 
-`Masthead` should contain the following components to maintain proper layout and formatting: `MastheadToggle`, `MastheadMain`, and `MastheadContent`.
+`Masthead` should contain the following components to maintain proper layout and formatting: `MastheadMain`, and `MastheadContent`.
```
Comment (edonehoo) [inline packages/react-core/src/components/Masthead/examples/Masthead.md:12]:
```suggestion
To maintain proper layout and formatting, a `<Masthead>` should contain both a `<MastheadMain>` and `<MastheadContent>` component.
```

File: packages/react-core/src/components/Masthead/examples/Masthead.md [hunk 3]
```diff
@@ -2,16 +2,16 @@
...
 
-`MastheadMain` represents the smaller area taken up by a logo, and will typically contain a `MastheadBrand`. `MastheadContent` represents the main portion of the masthead area and will typically contain a `Toolbar` or other menu-like content such as `Dropdown`.
+`MastheadMain` represents the smaller area taken up by a toggle and logo, and will typically contain a `MastheadToggle` and a `MastheadLogo`. `MastheadContent` represents the main portion of the masthead area and will typically contain a `Toolbar` or other menu-like content such as `Dropdown`.
```
Comment (edonehoo) [inline packages/react-core/src/components/Masthead/examples/Masthead.md:14]:
```suggestion
The `<MastheadMain>` component includes the smaller area that typically contains the `<MastheadToggle>` and a `<MastheadLogo>`. `<MastheadContent>` represents the main portion of the masthead, and will typically contain a `<Toolbar>` or other menu-like components, like a `<Dropdown>`.
```

File: packages/react-core/src/components/Page/examples/Page.md [hunk 4]
```diff
@@ -16,7 +16,7 @@ import c_page_section_m_limit_width_MaxWidth from '@patternfly/react-tokens/dist
 A page will typically contain the following components:
 
 - A `<Page>` with a `masthead` prop that often contains a [masthead](/components/masthead) component
-  - Mastheads contain the `<PageToggleButton>`, a `<MastheadMain>` that contains a `<MastheadBrand>`, and the page's header toolbar within `<MastheadContent>`.
+  - Mastheads contain the `<MastheadMain>` that wraps a `<PageToggleButton>` and `<MastheadBrand>`. The `<MastheadBrand>` wraps `<MastheadLogo>`. The masthead also contains  the page's header toolbar within `<MastheadContent>`.
```
Comment (edonehoo) [inline packages/react-core/src/components/Page/examples/Page.md:19]:
```suggestion

The `<MastheadMain>` component includes the smaller area that typically contains the `<MastheadToggle>` and a `<MastheadLogo>`. `<MastheadContent>` represents the main portion of the masthead, and will typically contain a `<Toolbar>` or other menu-like components, like a `<Dropdown>`.

  - Mastheads contain a `<MastheadMain>` component, which includes the `<MastheadToggle>`, a `<MastheadLogo>`,  and the page's toolbar (via `<MastheadContent>`.) The `<MastheadToggle>` component contains a `<PageToggleButton>`, and the `<MastheadLogo>` component contains a `<MastheadBrand>`. 
```

This is going off of the changes from the masthead page, where mention of MastheadBrand was replaced. But I may have misinterpreted that & you may need to correct this suggestion for accuracy

**Review by edonehoo — CHANGES_REQUESTED** [review]
since surge is down, this is going off the file changes as best I can. can re-review if/when surge comes back to make sure I checked everything, but this probably covers the main things regardless!

**Review by thatblindgeye — APPROVED** [review]

File: packages/react-integration/demo-app-ts/src/components/demos/MastheadDemo/MastheadDemo.tsx [hunk 5]
```diff
@@ -114,14 +115,16 @@ export class MastheadDemo extends Component<MastheadProps> {
...
     return (
       <Masthead id="basic">
-        <MastheadToggle>
-          <Button variant="plain" onClick={() => {}} aria-label="Global navigation">
-            <BarsIcon />
-          </Button>
-        </MastheadToggle>
         <MastheadMain>
+          <MastheadToggle>
+            <Button variant="plain" onClick={() => {}} aria-label="Global navigation">
```
[the remark this answers is not in the document]
Comment (tlabaj) [reply packages/react-integration/demo-app-ts/src/components/demos/MastheadDemo/MastheadDemo.tsx:120]: this is in the demo app for integration test and they are currently no test for masthead.  I don't think it is necessary to update this here.

**Review by edonehoo — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]
lgtm. We may want to open a follow up in org if we don't already have one to update the structure for the docs site (the surge masthead is the old structure still).

**Review by andrew-ronaldson — APPROVED** [review]
Great stuff!

**Review by srambach — APPROVED** [review]
🎉 ⭐

**Pull request merged**